src/sql/src/plan/statement/show.rs

```diff
@@ -479,13 +479,16 @@ pub fn show_indexes<'a>(
 
     let query = format!(
         "SELECT
+            idxs.name AS name,
+            objs.name AS on,
             clusters.name AS cluster,
-            objs.name AS on_name,
-            idxs.name AS key_name,
-            idx_cols.index_position AS seq_in_index,
-            obj_cols.name AS column_name,
-            idx_cols.on_expression AS expression,
-            idx_cols.nullable AS nullable
+            ARRAY_AGG(
+                CASE
+                    WHEN idx_cols.on_expression IS NULL THEN obj_cols.name
+                    ELSE idx_cols.on_expression
+                END
+                ORDER BY idx_cols.index_position ASC
+            ) AS key
```

Comment:
This query is honestly fairly slow on my computer, close to 1 second. I'm not sure what the best thing to do here is though. I don't think we should create a new builtin table with de-normalized index data. If the builtin views had indexes by default then creating a builtin view for this might help.

Reply:
Have you tried in release mode? I bet it's fast enough that we no longer feel bad about its speed.